tests/tests_unit/test_base.py

```diff
@@ -250,6 +267,44 @@ def test_load_has_no_side_effects(
         # Ensure no part of load mutates the input dict:
         assert dumped == original_dumped
 
+    # These properties are guaranteed to not change from object in the API,
```

Comment:
Not quite following this. Any object in the spec will remain an object, why won't we inject extra keys into these particular ones?

Reply:
The tests below dumps all CogniteObject instances and injects new parameters in any dictionary it finds in the dumped object, including in nested structures. If you do this in a container object, you will find the properties dictionary and inject into than, which is the reason for all the previous `isinstance(, dict)` checks. 

These are the exceptions where we are not injecting an arbitrary parameter to check that it does not break `_load(...)`.